src/test/java/weizberg/gameoflife/GameOfLifeControllerTest.java

```diff
@@ -0,0 +1,189 @@
+package weizberg.gameoflife;
+
+import org.apache.commons.io.IOUtils;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import static org.mockito.Mockito.*;
+
+public class GameOfLifeControllerTest {
+
+    @Test
+    void toggleCellOn() {
+        //given
+        Grid model = mock();
+        GridComponent view = mock();
+        final GameOfLifeController controller = new GameOfLifeController(model, view);
+        doReturn(10).when(view).getCellSize();
+        doReturn(10).when(model).getWidth();
+        doReturn(10).when(model).getHeight();
+
+        //when
+        controller.toggleCell(50, 100);
+
+        //then
+        verify(model).makeAlive(5, 10);
+        verify(view).repaint();
+    }
+
+    @Test
+    void toggleCellOff() {
+        //given
+        Grid model = mock();
+        GridComponent view = mock();
+        final GameOfLifeController controller = new GameOfLifeController(model, view);
+        doReturn(10).when(view).getCellSize();
+        doReturn(10).when(model).getWidth();
+        doReturn(10).when(model).getHeight();
+        doReturn(true).when(model).isAlive(5, 10);
+
+        //when
+        controller.toggleCell(50, 100);
+
+        //then
+        verify(model).makeDead(5, 10);
+        verify(view).repaint();
+    }
+
+
+    @Test
+    public void pasteUrl() throws IOException {
+
+        //Given
+        Grid model = mock();
+        GridComponent view = mock();
+        RleParser rleParser = mock();
+        final GameOfLifeController controller = new GameOfLifeController(model, view, rleParser);
+
+        int[][] mockedField = new int[100][100];
+        mockedField[49][50] = 1;
+        mockedField[50][51] = 1;
+        mockedField[51][49] = 1;
+        mockedField[51][50] = 1;
+        mockedField[51][51] = 1;
+
+        //When
+        String url = "https://conwaylife.com/patterns/glider.rle";
+        when(rleParser.rleToField(anyString())).thenReturn(mockedField);
+
+        InputStream in = new URL(url).openStream();
+        String contentOfUrl = IOUtils.toString(in);
+
+        controller.paste(contentOfUrl);
+
+        // Then
+        verify(model).setField(argThat(field -> {
```

Comment:
instead of doing this cell by cell, make a method that you can pass a whole 2d array.